management/src/clusterm/manager/api.go

```diff
@@ -141,36 +137,35 @@ func validateAndSanitizeEmptyExtraVars(errorPrefix, extraVars string) (string, e
 	return extraVars, nil
 }
 
-func (m *Manager) nodesCommission(tags []string, sanitizedExtraVars string) error {
-	me := newWaitableEvent(newCommissionEvent(m, tags, sanitizedExtraVars))
+func (m *Manager) nodesCommission(req *APIRequest, extraVars string) error {
+	if !IsValidHostGroup(req.HostGroup) {
+		return errored.Errorf("host-group is a mandatory parameter and is not specified %s", req.HostGroup)
+	}
+	me := newWaitableEvent(newCommissionEvent(m, req, extraVars))
```

Comment:
I think we should keep the `new...Event()` functions continue to explicitly take the arguments as part of the requests that they are interested in. This keeps the event interface contract explicit without having to look inside `new...Event()` function itself.